service/collector/key/key.go

```diff
@@ -4,11 +4,6 @@ import (
 	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 )
 
-// ClusterID returns the unique ID for this cluster.
-func ClusterID(customObject providerv1alpha1.AzureConfig) string {
```

Comment:
This is weird. I mean, it's ok if it's not used, but I remember from `aws-operator` Node Pools refactoring that `ClusterID()` was very widely used everywhere. I guess we'll find this out when we start doing similar refactoring here :slightly_smiling_face:

Reply:
I guess this makes sense now that I know you had azure-operator in mind.

Reply:
Yes, definitely :stuck_out_tongue: